src/eval/reduction.rs

```diff
@@ -936,9 +936,12 @@ fn apply_continuation<F: LurkField>(
                     Op1::Char => match result.tag {
                         ExprTag::Num | ExprTag::Char => {
                             let scalar_ptr = store
-                                .get_expr_hash(&result)
+                                .hash_expr(&result)
                                 .ok_or_else(|| store::Error("expr hash missing".into()))?;
-                            store.get_char_from_u32(scalar_ptr.value().to_u32_unchecked())
+                            eprintln!("char {:?}", scalar_ptr.value());
+                            store.intern_char(
+                                char::from_u32(scalar_ptr.value().to_u32_unchecked()).unwrap(),
```

Comment:
`LurkField::to_u32_unchecked()` has the effect of truncating a field element to 32 bits, so we can't fail on larger values. We *can* fail on u32 values which are not valid Rust `char`s (cf. https://doc.rust-lang.org/std/primitive.char.html#method.from_u32). 

This case is not currently handled on `master`, but I've added a check here which returns a `Control::Error` on these cases rather than panicking, but conceivably this should be some other kind of error in the future